libarchive/archive_read_support_format_zip.c

```diff
@@ -1174,17 +1227,45 @@ zip_read_local_file_header(struct archive_read *a, struct archive_entry *entry,
 			return ARCHIVE_FATAL;
 		}
 
+    size_t linkname_full_length = linkname_length; // take into account compression if any
```

Comment:
Rest of the file uses tabs as indentation. Maybe it would be a good idea to re-tab the commit and use tabs as well?

Reply:
done